C error: Calling enum function with argument as Option of Sum when match is struct

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.10 080684d, press to see full `v doctor` output</summary>

|V full version      |V 0.4.10 7dc3889f19e7bdc6fb803ad4d1d65fd9d50c6e0c.080684d
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 22.04.5 LTS
|Processor           |4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-4150 CPU @ 3.50GHz
|Memory              |0.29GB/3.7GB
|                    |
|V executable        |/home/jorge/v/v
|V last modified time|2025-06-17 22:47:06
|                    |
|V home dir          |OK, value: /home/jorge/v
|VMODULES            |OK, value: /home/jorge/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/jorge/bugs
|                    |
|Git version         |git version 2.34.1
|V git status        |weekly.2025.16-311-g080684dd
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
|gcc version         |gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
|clang version       |Ubuntu clang version 14.0.0-1ubuntu1.1
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.5 ()
|glibc version       |ldd (Ubuntu GLIBC 2.35-0ubuntu3.10) 2.35

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  250617.v && 250617`
```v
type TxPayload = Can | u8

enum Tx {
	empty
	data
	can
}

fn (tx Tx) frame_bytes(payload ?TxPayload) ! []u8 {
	match tx {
		.empty {
			return [ u8(tx) ]
		}
		.data {
			if payload != none {
				if payload is u8 {
					return [ u8(tx), payload ]
				}		
			}
			return error('Invalid data')
		}
		.can {
			if payload != none {
				if payload is Can {
					return [ u8(tx), payload.net ] // fails with this line
				}
			}
			return error('Invalid can')
		}
	}
}

struct Can {
	net  u8
}

fn main()  {
	Tx.empty.frame_bytes(none)!
}

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/250617.01JY02HM2SSEHC2FVKR88KY03Z.tmp.c:5490: error: struct or union expected
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

Compile

`-show-c-output` message:

```
=================================================================================
======== Output of the C Compiler (cc) ========
/tmp/v_1000/250617.01JY02EK6BE8Y0K0W6NFJEA6N1.tmp.c: In function ‘main__Tx_frame_bytes’:
/tmp/v_1000/250617.01JY02EK6BE8Y0K0W6NFJEA6N1.tmp.c:5490:182: error: ‘((main__TxPayload *)&payload.data)-><U3428>._main__Can’ is a pointer; did you mean to use ‘->’?
 5490 |                                 _result_ok(&(Array_u8[]) { new_array_from_c_array_noscan(2, 2, sizeof(u8), _MOV((u8[2]){((u8)(tx)), *((*(main__TxPayload*)(payload.data))._main__Can).net})) }, (_result*)(&_t4), sizeof(Array_u8));
      |                                                                                                                                                                                      ^
      |                                                                                                                                                                                      ->
===============================================

```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -5287,8 +5287,9 @@ fn (mut g Gen) ident(node ast.Ident) {
 				obj_sym := g.table.sym(g.unwrap_generic(node.obj.typ))
 				if !prevent_sum_type_unwrapping_once {
 					nested_unwrap := node.obj.smartcasts.len > 1
-					if is_option && nested_unwrap && obj_sym.kind == .sum_type {
-						g.write('*(')
+					unwrap_sumtype := is_option && nested_unwrap && obj_sym.kind == .sum_type
+					if unwrap_sumtype {
+						g.write('(*(')
 					}
 					for i, typ in node.obj.smartcasts {
 						is_option_unwrap := i == 0 && is_option
@@ -5378,6 +5379,9 @@ fn (mut g Gen) ident(node ast.Ident) {
 								} else if !is_option_unwrap
 									&& obj_sym.kind in [.sum_type, .interface] {
 									g.write('${dot}_${cast_sym.cname}')
+								}
+								if i != 0 && unwrap_sumtype {
+									g.write(')')
 								}
 							}
 						}

--- a/vlib/v/tests/options/option_sumtype_unwrap_test.v
+++ b/vlib/v/tests/options/option_sumtype_unwrap_test.v
@@ -1,0 +1,40 @@
+type TxPayload = Can | u8
+
+enum Tx {
+	empty
+	data
+	can
+}
+
+fn (tx Tx) frame_bytes(payload ?TxPayload) ![]u8 {
+	match tx {
+		.empty {
+			return [u8(tx)]
+		}
+		.data {
+			if payload != none {
+				if payload is u8 {
+					return [u8(tx), payload]
+				}
+			}
+			return error('Invalid data')
+		}
+		.can {
+			if payload != none {
+				if payload is Can {
+					return [u8(tx), payload.net]
+				}
+			}
+			return error('Invalid can')
+		}
+	}
+}
+
+struct Can {
+	net u8
+}
+
+fn test_main() {
+	assert Tx.empty.frame_bytes(none)! == [u8(0)]
+	assert Tx.data.frame_bytes(u8(123))! == [u8(1), 123]
+}

